packages/ic-http-certification/src/utils/response_header.rs

```diff
@@ -0,0 +1,95 @@
+use crate::{HttpResponse, CERTIFICATE_HEADER_NAME};
+use base64::{engine::general_purpose::STANDARD as BASE64, Engine};
+use ic_certification::HashTree;
+use serde::Serialize;
+
+/// Adds the `IC-Certificate` header to a given [`HttpResponse`]. This header is used by the HTTP Gateway
+/// to verify the authenticity of query call responses.
+///
+/// # Arguments
+///
+/// * `data_certificate` - A certificate used by the HTTP Gateway to verify a response.
+///     Retrieved using `ic_cdk::api::data_certificate`.
+/// * `response` - The [`HttpResponse`] to add the certificate header to.
+///     Created using [`HttpResponse::builder()`](crate::HttpResponse::builder).
+/// * `witness` - A pruned merkle tree revealing the relevant certification for the current response.
+///     Created using [`HttpCertificationTree::witness()`](crate::HttpCertificationTree::witness).
+/// * `expr_path` - An expression path for the current response informing the HTTP Gateway where the
+///     relevant certification is present in the merkle tree. Created using
+///     [`HttpCertificationPath::to_expr_path()`](crate::HttpCertificationPath::to_expr_path).
+///
+/// # Example
```

Comment:
I guess the example will be executed by CI, so in a way it it doubles as a unit test, (and also there are examples in this PR that use this function), but maybe it would be worthwhile to add a test explicitly, just to be sure about the test coverage?

Reply:
Indeed the examples are run by the CI pipeline and can serve as both documentation and unit tests. We can already be sure that the test covers the execution of the function, so I'm not sure if duplicating effort with additional unit tests would provide any value. Am I perhaps missing something?

Reply:
Discussed offline and agreed to leave as is.